src/js/inject.js

```diff
@@ -465,6 +511,151 @@ $('body').on('click', '#open-modal', (ev) => {
   }
 });
 
+// http://stackoverflow.com/a/2091331
+const getQueryVariable = (str, variable) => {
+  const vars = str.substring(1).split('&');
+  for (let i = 0; i < vars.length; i += 1) {
+    const pair = vars[i].split('=');
+    if (decodeURIComponent(pair[0]) === variable) {
+      return decodeURIComponent(pair[1]);
+    }
+  }
+  return null;
+};
+
+const setClipboard = (text) => {
+  const tc = $('.compose-text-container .js-compose-text');
+  const orig = tc.val();
+  const active = document.activeElement;
+  tc.val(text);
+  tc[0].focus();
+  tc[0].setSelectionRange(0, text.length);
+  document.execCommand('copy');
+  tc.val(orig);
+  active.focus();
+};
+
+const cleanMediaUrl = (str) => {
+  return str
+    .replace(/url\(/g, '')
+    .replace(/["')]/g, '')
+    .replace(/:(thumb|small|medium|large|orig)$/, '')
+    .replace(/^none$/, '');
+};
+
+const getMediaFromEmbed = (embed) => {
+  const iframeSrc = $(embed).find('iframe').attr('src');
+  if (iframeSrc && iframeSrc.length) {
+    return getQueryVariable(iframeSrc, 'video_url');
+  }
+  return '';
+};
+
+const getMedia = (elem) => {
+  const media = [];
+
+  // check the obscure case of being in a modal
+  const modalContainer = $(elem).parents('.js-modal-panel');
+
+  $(modalContainer).find('.js-embeditem').each((i, embed) => {
+    const iframeSrc = getMediaFromEmbed(embed);
+    if (iframeSrc.length) {
+      media.push(iframeSrc);
+    }
+  });
+  if (media.length) return media;
+
+  // we want to be sure this modal is a gif
+  $(modalContainer).parents('[data-btd-provider="gif"]')
+    .find('.btd-embed-container video').each((i, link) => {
+      const vidSrc = $(link).attr('src');
+      if (vidSrc.length) {
+        media.push(vidSrc);
+      }
+    });
+  if (media.length) return media;
+
+  // check the most typical case of a stream item
+  const normalContainer = $(elem).parents('article');
+
+  $(normalContainer).find('.quoted-tweet').each((i, link) => {
+    const tweetID = $(link).data('tweet-id');
+    const fullURI = `${$(link).find('.account-link').attr('href')}/status/${tweetID}`;
```

Comment:
That seems risky to put a jQuery expression inside your string, I'd put this in a variable and add a failover in case this fails for whatever reason if TweetDeck changes their layout or whatnot.